src/components/TokenTable.js

```diff
@@ -0,0 +1,102 @@
+import React, { useState, useCallback } from 'react';
+import { useStaticQuery, graphql } from 'gatsby';
+import { SprkTable, SprkTextInput } from '@sparkdesignsystem/spark-react';
+import debounce from 'lodash/debounce';
+
+const TokenTable = () => {
+  const tokenData = useStaticQuery(
+    graphql`
+      query SassVars {
+        allSassVarsJson {
+          edges {
+            node {
+              description
+              context {
+                name
+                value
+              }
+            }
+          }
+        }
+      }
+    `,
+  );
+  const sassVarData = tokenData.allSassVarsJson.edges.map((item) => ({
+    name: item.node.context.name,
+    value: item.node.context.value,
+    description: item.node.description,
+    flattenedData: [
+      item.node.context.name,
+      item.node.context.value,
+      item.node.description,
+    ]
+      .join(' ')
+      .toLowerCase(),
+  }));
+
+  const [filteredData, setFilteredData] = useState(sassVarData);
+  const [value, setValue] = useState('');
+
+  const debouncedFilterData = useCallback(
+    debounce((searchQuery) => {
+      setFilteredData(
+        sassVarData.filter(({ flattenedData }) => {
+          return flattenedData.includes(searchQuery.toLowerCase());
+        }),
+      );
+    }, 500),
+    [],
+  );
+
+  const handleChange = (event) => {
+    const { value: searchQuery } = event.target;
+    setValue(searchQuery);
+    debouncedFilterData(searchQuery);
+  };
+
+  return (
+    <>
+      <SprkTextInput
+        label="Filter Through Token List:"
+        name="text-input-label"
+        value={value}
+        onChange={handleChange}
+      />
+      {filteredData.length <= 0 ? (
+        <p>
+          No results for
+          <span className="sprk-u-FontStyle--italic">
+            &nbsp;&lsquo;{value}&rsquo;
```

Comment:
does {value} need to be sanitized? might just want to remove writing it out to the dom

Reply:
Talking with design, it's a good idea to provide feedback as to what the user searched for. In terms of sanitized you mean, do a `toString()` on it?